lib/libutils/ext/trace.c

```diff
@@ -117,39 +117,26 @@ void trace_printf(const char *function, int line, int level, bool level_ok,
 		return;
 	boffs += res;
 
-	/* Print the core ID if in atomic context  */
-	if (level_ok && (BIT(level) & CFG_MSG_LONG_PREFIX_MASK))
+	if (level_ok && (BIT(level) & CFG_MSG_LONG_PREFIX_MASK)) {
+		/* Print the core ID if in atomic context  */
 		res = print_core_id(buf + boffs, sizeof(buf) - boffs);
-	else
-		res = 0;
-	if (res < 0)
-		return;
-	boffs += res;
+		if (res < 0)
+			return;
+		boffs += res;
 
-	/* Print the Thread ID */
-	if (level_ok && !(BIT(level) & CFG_MSG_LONG_PREFIX_MASK))
-		thread_id = -1;
-	else
+		/* Print the Thread ID */
 		thread_id = trace_ext_get_thread_id();
```

Comment:
For consistency with `print_core_id()` you might want to move `trace_ext_get_thread_id()` into `print_thread_id()`.